File: KoRadio/UI/ko_radio_mobile/lib/models/employee_task.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:ko_radio_mobile/models/company_employee.dart';

part 'employee_task.g.dart';

@JsonSerializable()
class EmployeeTask {
  int employeeTaskId;
  String? task;
  bool? isFinished;
  int? companyEmployeeId;
  DateTime? createdAt;
  int? jobId;
  int? companyId;
  CompanyEmployee? companyEmployee;

  EmployeeTask({required this.employeeTaskId});
    factory EmployeeTask.fromJson(Map<String, dynamic> json) => _$EmployeeTaskFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$EmployeeTaskToJson(this);

}

File: KoRadio/KoRadio.Model/EmployeeTask.cs
using KoRadio.Model.DTOs;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KoRadio.Model
{
	public class EmployeeTask
	{
		public int EmployeeTaskId { get; set; }

		public string Task { get; set; } = null!;

		public bool IsFinished { get; set; }

		public int? CompanyEmployeeId { get; set; }
		public DateTime CreatedAt { get; set; }
		public int JobId { get; set; }

		public int CompanyId { get; set; }

		public virtual Company Company { get; set; } = null!;

		public virtual CompanyEmployeeDto? CompanyEmployee { get; set; }

		public virtual Job Job { get; set; } = null!;

	
	}
}
